Add Code_Aster OpenFOAM Tutorial

micheltakken opened the pull request:
In this branch the files for the flow-over-plate tutorial have been added. The tutorial should work after cloning this, and following the description on the wiki for the OpenFOAM-Code_Aster flow-over-plate tutorial

MakisH commented on tutorials/buoyantSimpleFoam-aster/fluid/system/changeDictionary-1:
```diff
@@ -0,0 +1,32 @@
+// Re = 500, Pr = 0.01, k = 1
```
Do we still need this file? It would only make sense for parametric studies. Do we refer to it?

micheltakken replied:
No we don't need it, it runs perfectly without. I have deleted these changeDictionary files.

MakisH commented on tutorials/buoyantSimpleFoam-aster/fluid/system/changeDictionary-10:
```diff
@@ -0,0 +1,32 @@
+// Re = 10000.0, Pr = 100, k = 1
```
Same for this and all other `changeDictionary` files.

micheltakken replied:
Deleted.

MakisH commented on tutorials/buoyantSimpleFoam-aster/precice-config.xml:
```diff
@@ -0,0 +1,56 @@
+<precice-configuration>
+  <log>
+    <sink enabled="true" output="stdout" type="stream" filter= "(%Severity% >= trace)"/>
+	<sink type="file" output="debug.log" filter="(%Severity% >= trace)" enabled="true" />
```
At the end, do we need this additional sink? If yes, add a comment.

micheltakken replied:
No we didn't need the additional sink. I have removed it.

MakisH commented on tutorials/buoyantSimpleFoam-aster/precice-config.xml:
```diff
@@ -0,0 +1,56 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<precice-configuration>
+  <log>
+    <sink enabled="true" output="stdout" type="stream" filter= "(%Severity% >= trace)"/>
```
Please use this updated format: https://github.com/precice/tutorials/blob/develop/FSI/flap_perp/OpenFOAM-CalculiX/precice-config_serial.xml#L6

micheltakken replied:
Format is updated.

MakisH commented on tutorials/buoyantSimpleFoam-aster/runFluid.sh:
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+    cd fluid; blockMesh; cd ..
+    buoyantSimpleFoam -case fluid #> fluid.log
```
Remove comment

micheltakken replied:
Removed.

MakisH commented on tutorials/buoyantSimpleFoam-aster/runSolid.sh:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
```
Could you add a few comments on what is going on here?

micheltakken replied:
The `as_run --quick` command is obsolete, and is removed.
I think the other two lines are quite self-explanatory, let me know if you still believe they should have a few comments.

MakisH replied:
Have you already pushed the latest version? I still see `as_run --quick` here. The other two are fine.

micheltakken replied:
Yes, this line of code is gone in the `tutorial-addition` branch ([link](https://github.com/precice/code_aster-adapter/blob/tutorial-addition/tutorials/buoyantSimpleFoam-aster/runSolid.sh))

MakisH commented on tutorials/buoyantSimpleFoam-aster/solid.log:
```diff
@@ -0,0 +1,45 @@
+<INFO> This functionnality is still in a beta state of development and may be removed a future release, or may never be improved. Please use --silent option to ignore this message the next time.
```
log file - remove

micheltakken replied:
Removed.

MakisH commented on tutorials/buoyantSimpleFoam-aster/solid/solid.export:
```diff
@@ -0,0 +1,44 @@
+P actions make_etude
+P aster_root /home/michel/Desktop/Code_Aster-14.4
```
hard-coded path

micheltakken replied:
Opened issue #7

MakisH commented on tutorials/buoyantSimpleFoam-aster/solid/solid.export:
```diff
@@ -0,0 +1,44 @@
+P corefilesize unlimited
+P cpresok RESNOOK
+P debug nodebug
+P display michel-HP-ZBook-Studio-G4.domain.org:0
```
more hard-coded things.

micheltakken replied:
Opened a separate issue for this. #10

micheltakken commented:
Following changes have been applied:
* Removed `fluid/constant/polyMesh` folder. `blockMeshDict` is now in `fluid/systems/`
* Removed all .gz files
* Removed `fluid/runFluid.sh`
* Removed `.mess` and `.resu` files. `.medd` defines the solid mesh for Code_Aster, and is needed.

micheltakken commented on tutorials/buoyantSimpleFoam-aster/config.comm:
```diff
@@ -0,0 +1,34 @@
+settings = \
+{'base-path': '/home/michel/Git/code_aster-adapter/tutorials/buoyantSimpleFoam-aster',
```
This is documented [here](https://github.com/precice/code_aster-adapter/wiki/Flow-over-plate-Code_Aster-Tutorial) under "Running the Test case". 
Ideally this path can be imported from the environment (set in bashrc). I have opened issue #7 for this

micheltakken commented on tutorials/buoyantSimpleFoam-aster/config.yml.org:
```diff
@@ -0,0 +1,33 @@
+base-path: "/home/michel/Desktop/Code_Aster-Adapter-Dummy/buoyantSimpleFoam-aster"
```
Opened issue #7

micheltakken commented on tutorials/buoyantSimpleFoam-aster/config.yml.org:
```diff
@@ -0,0 +1,33 @@
+base-path: "/home/michel/Desktop/Code_Aster-Adapter-Dummy/buoyantSimpleFoam-aster"
+
```
The coupling also runs without this file. 

 `tutorials/buoyantSimpleFoam-aster/config.yml.org` can be removed

Opened issue #8

micheltakken commented on tutorials/buoyantSimpleFoam-aster/fluid.log:
```diff
@@ -0,0 +1,166 @@
+/*---------------------------------------------------------------------------*\
```
Removed.

micheltakken commented on tutorials/buoyantSimpleFoam-aster/debug.log:
```diff
@@ -0,0 +1,43 @@
+(0) 14:54:13 [impl::SolverInterfaceImpl]:91 in configure: [0mThis is preCICE version 1.4.0
```
Removed.

micheltakken commented on tutorials/buoyantSimpleFoam-aster/fluid/Allclean:
```diff
@@ -0,0 +1,2 @@
+. $WM_PROJECT_DIR/bin/tools/CleanFunctions
+cleanCase
```
Opened issue #9

micheltakken commented on tutorials/buoyantSimpleFoam-aster/fluid/0/U:
```diff
@@ -0,0 +1,64 @@
+/*--------------------------------*- C++ -*----------------------------------*\
```
Removed the headers from every OpenFOAM file.

micheltakken replied:
Removed these headers.

micheltakken commented on tutorials/buoyantSimpleFoam-aster/fluid/0/T:
```diff
@@ -0,0 +1,75 @@
+    interface
+    {
+        type            mixed;
+        refGradient		uniform 0;
```
Fixed.

micheltakken commented on tutorials/buoyantSimpleFoam-aster/fluid/0/U:
```diff
@@ -0,0 +1,64 @@
+    top
+    {
+    	type			slip;
+        value           uniform (.1 0 0);
```
Fixed.

MakisH commented on tutorials/buoyantSimpleFoam-aster/precice-config.xml:
```diff
@@ -0,0 +1,55 @@
+      <read-data mesh="Fluid-Interface" name="Heat-Transfer-Coefficient-Solid"/>
+      <mapping:nearest-neighbor constraint="consistent" direction="read" to="Fluid-Interface" from="Solid-Interface-Nodes"/>
+    </participant>
+    <m2n:sockets to="Fluid" from="Solid" exchange-directory="/home/michel/Desktop/Code_Aster-Adapter-Dummy/buoyantSimpleFoam-aster" distribution-type="gather-scatter"/>
```
We don't / should not need this hard-coded path. It defaults to the current directory.

```suggestion
    <m2n:sockets to="Fluid" from="Solid" distribution-type="gather-scatter"/>
```

micheltakken replied:
Both adapters are unable to setup a master communication to the coupling partner when this change is implemented.
